SlicerDiffusionQC/GradQC.py

```diff
@@ -330,38 +335,54 @@ def onApplyButton(self):
 
     # if results don't exist or autoMode is specified, run cli-module, this may overwrite previous results
     if not success or not self.autoMode: # (not self.autoMode) = True when check box unchecked
-
-        diffusionQCcli = slicer.modules.diffusionqc
-        cliNode = slicer.cli.run(diffusionQCcli, parameters=parameters)
-
-        # Track progress of the algorithm
-        self.progressBar.setCommandLineModuleNode(cliNode)
-
-        def observeStatus(caller,_):
-          # Check if CLI completed w/o any error
-          if caller.GetStatusString()=='Completed' and cliNode.GetErrorText()=='':
-
-            # If mask was not given, created in the pipeline, load now
-            if self.createMask:
-
-              # Determine prefix and directory
-              directory = os.path.dirname(os.path.abspath(self.inputSelector.currentPath))
-              prefix = os.path.basename(self.inputSelector.currentPath.split('.')[0])
-
-              self.maskSelector.currentPath= os.path.join(directory, prefix+'_mask'+'.nrrd')
-              self.onSelectMask()
-
-            self.GUI()
-
-        cliNode.AddObserver('ModifiedEvent', observeStatus)
+    
+    '''
+     
+    # Determine prefix and directory
+    if self.outputDirSelector.currentPath is None:
+        directory = os.path.dirname(os.path.abspath(dwiPath))
+        self.outputDirSelector.currentPath= directory
+    else:
+        directory = self.outputDirSelector.currentPath
+        if not os.path.exists(directory):
+            os.mkdir(directory)
+
+    prefix = os.path.basename(self.inputSelector.currentPath.split('.')[0])
+    
+    if self.createMask:
+        self.maskSelector.currentPath= os.path.join(directory, prefix+'_mask'+'.nrrd')        
+    
+    diffusionQCcli = slicer.modules.diffusionqc
+    
+    # Slicer 4.11
+    cliNode= slicer.cli.run(diffusionQCcli, parameters)
+
+    # Slicer 4.9    
+    # cliNode= slicer.cli.run(diffusionQCcli, None, parameters)
+    
```

Comment:
Tashrif's local computer has Slicer 4.9 but may be Isaiah's computer had 4.11. That's why this issue arose before. However, Tashrif developed originally on Slicer 4.9 with `None`